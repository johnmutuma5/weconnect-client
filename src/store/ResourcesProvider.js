import React from 'react';
import PropTypes from 'prop-types';

class ResourcesProvider extends React.Component {
    getChildContext() {
        return {
            store: this.props.store
        }
    }

    render() {
        return this.props.children;
    }
}


ResourcesProvider.childContextTypes = {
    store: PropTypes.object.isRequired
}

export default ResourcesProvider;
