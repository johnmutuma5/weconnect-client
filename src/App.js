import React, { Component } from 'react';
import Layout from './containers/Layout/Layout';
import { Route } from 'react-router-dom';
import { Switch } from 'react-router';
import Businesses from './containers/Businesses/Businesses';
import BusinessesFilter from './containers/BusinessesFilter/BusinessesFilter';
import SearchBusinesses from './containers/Search/Search';
import UserRegistrationForm from './containers/Auth/UserRegistration/UserRegistrationForm';
import BusinessProfile from './containers/BusinessProfile/BusinessProfile';
import ResourcesProvider, { createStore } from './store';
import weConnectReducer from './store/baseReducer';


export const store = createStore(weConnectReducer);

class App extends Component {
  render() {
    return (
        <ResourcesProvider store={ store }>
            <Layout>
                <Switch>
                    <Route exact path="/" component={ Businesses } />
                    <Route exact path="/businesses" component={ Businesses } />
                    <Route exact path="/businesses/filter" component={ BusinessesFilter } />
                    <Route exact path="/businesses/search" component={ SearchBusinesses } />
                    <Route exact path="/businesses/:id" component={ BusinessProfile } />
                </Switch>
            </Layout>
        </ResourcesProvider>
    );
  }
}

export default App;
