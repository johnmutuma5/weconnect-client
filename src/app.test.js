import App from './App';
import ResourcesProvider, { createStore } from './store';
import weConnectReducer from './store/baseReducer';

describe('<App />', () => {
    test('global testing variables ok', () => {
        /// sanity test
        expect.assertions(3);
        expect(shallow).not.toBeNull();
        expect(React).not.toBeNull();
        expect(localStorage).not.toBeNull();
    });

    test('ResourcesProvider renders <App />', () => {
        const store = createStore(weConnectReducer);

        const wrapper = shallow(<ResourcesProvider store={store}>
                                    <App />
                                </ResourcesProvider>);
        expect(wrapper).toMatchSnapshot();
    });

    test('App renders correctly', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    })
})
