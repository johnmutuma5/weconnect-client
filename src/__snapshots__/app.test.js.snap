// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App renders correctly 1`] = `
<Layout>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/businesses"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/businesses/filter"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/businesses/search"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/businesses/:id"
    />
  </Switch>
</Layout>
`;

exports[`<App /> ResourcesProvider renders <App /> 1`] = `<App />`;
